### ISSUE TITLE
o-forms: add radio or button role to pseudo-radio-link

### ISSUE THREAD

**Issue by JakeChampion** [Financial-Times/origami#535]
This was mentioned in #532

> Additionally, emulating links with radio buttons maybe confusing for users of assistive technologies and should be given an appropriate role of radio or button as well as a shared label using a fieldset and legend. Further information can be found in the Radios section of the GOV.UK Design System.

**Comment by notlee** [comment]
This one will need some thought and thorough testing. Will a link still work as a link with a new role. Would it be more confusing to present as a form element and work as a link? Do we update products to use the standard square radio buttons instead and deprecate this? Or do we need to change the design?

Context, this was originally added for this myft ui: https://github.com/Financial-Times/o-forms/pull/225

**Comment by notlee** [comment]
Looks like this is pretty much just used on the myft feed page and, especially given its complexity, I would be keen to deprecate `pseudo-radio-link`.

![Screenshot 2023-01-05 at 09 35 20](https://user-images.githubusercontent.com/10405691/210748188-124711d8-0e9c-42a6-bd52-f996d73700c8.png)

On the myft page there is nothing linking the "View" label with the topic/timeline `pseudo-radio-link` elements. Though they look like a form input they behave entirely like links and have `aria-current="page"` attributes accordingly:
```
<div class="feed-toolbar__toggle">
    <div class="feed-toolbar__toggle-label">
        View
    </div>
    <div class="o-forms-input--pseudo-radio-link o-forms-input--feed-toolbar">
        <a class="o-forms-input__link o-forms-input__link--grid-icon o-forms-input__link--current feed-toolbar__toggle-label--link" href="#" aria-current="page" data-trackable="topic">Topic</a>
        <a class="o-forms-input__link o-forms-input__link--list-icon  feed-toolbar__toggle-label--link" href="#" data-trackable="time">Timeline</a>
    </div>
</div>
```

**Comment by notlee** [comment]
☝️ the above is a silly idea, as the buttons don't present as an exclusive option (it would appear as though both can be pressed at the same time, which they cannot). In this case a standard radio box which submits on change would be best, with submit button available for the core experience.
